doc/api/esm.md

```diff
@@ -239,11 +241,33 @@ const { default: barData } =
   await import('./bar.json', { assert: { type: 'json' } });
 ```
 
-Node.js supports the following `type` values:
+Node.js supports the following `type` values, for which the assertion is
+mandatory:
+
+| Assertion `type` | Needed for       |
+| ---------------- | ---------------- |
+| `'json'`         | [JSON modules][] |
+| `'webassembly'`  | [Wasm modules][] |
+
+Imports of CommonJS and ES module JavaScript files, and Node.js core modules, do
+not need an import assertion; but they can be given a `type` of `'javascript'`
+if desired:
+
+```js
+const url = getUrlOfModuleToBeDynamicallyImported();
+const { pathname } = new URL(url); // Exclude `data:`, query strings and hashes
+
+let assertType = 'javascript';
+if (pathname.endsWith('.json') ||
+    pathname.startsWith('application/json')) {
```

Comment:
Something like `data:text/javascript,export{}//.json` would be picked up as JSON, when it's actually valid JS. Also, should we add a second parameter to the `URL` constructor?
```suggestion
const { protocol, pathname } = new URL(url, import.meta.url);

let assertType = 'javascript';
if (protocol === 'data:' ?
      pathname.startsWith('application/json') :
      pathname.endsWith('.json')) {
```
Same thing for `.wasm` check.